app/src/main/java/com/github/polybooks/FilteringBooksActivity.kt

```diff
@@ -0,0 +1,145 @@
+package com.github.polybooks
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Button
+import android.widget.CheckBox
+import android.widget.RadioButton
+import android.widget.RadioGroup
+import androidx.appcompat.app.AppCompatActivity
+import com.github.polybooks.core.Course
+import com.github.polybooks.core.Field
+import com.github.polybooks.core.Interest
+import com.github.polybooks.core.Semester
+import com.github.polybooks.core.database.implementation.DummyBookQuery
+import com.github.polybooks.core.database.implementation.DummySalesQuery
+import com.github.polybooks.core.database.interfaces.BookOrdering
+import com.github.polybooks.core.database.interfaces.BookQuery
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * This activity let the users to select the sorting and filtering parameters
+ * by clicking on the corresponding value (eg. "Architecture" button for
+ * parameter "Field"). If no value is selected for a given parameter, then the
+ * query won't be filtered on this parameter
+ *
+ * Note : should implement a dynamic version of it soon
+ */
+class FilteringBooksActivity : AppCompatActivity() {
+
+    private lateinit var mResetParameters : Button
+    private lateinit var mResults : Button
+
+    //--- TODO hardcoded parameters: make it dynamic
+    private lateinit var mSortGroup : RadioGroup
+    private lateinit var mSortPopularity : RadioButton
+    private lateinit var mSortTitleInc : RadioButton
+    private lateinit var mSortTitleDec : RadioButton
+
+    private lateinit var mFieldCS : CheckBox
+    private lateinit var mFieldBio : CheckBox
+    private lateinit var mFieldArchi : CheckBox
+
+    private lateinit var mSemBa1 : CheckBox
+    private lateinit var mSemBa2 : CheckBox
+    private lateinit var mSemBa3 : CheckBox
+    private lateinit var mSemMa1 : CheckBox
+    private lateinit var mSemMa2 : CheckBox
+
+    private lateinit var mCourseCS306 : CheckBox
+    private lateinit var mCourseCOM480 : CheckBox
```

Comment:
Cf le xml, ça devrait plutôt être 3 Spinners

Reply:
Good point, I was initially thinking that spinners would let us only check one parameter from the set (that was the initial reason why I chosed checkbox). But after some research it appears that I could make an hybrid version Spinner-Checkbox, it doesn't seem to be something straightforward neither common though, I suggest to implement it as a part of another sprint

Reply:
Oh, I get what to mean, I didn't thought about letting the user selecting several courses at once.
But even without doing that, I think it's much less harmful for the user to force them to search one course at the time (which will probably be their usual flow anyways), than to present them with a page with literally over hundreds of checkboxes...
Fine with it being a task for another sprint, but let's not forget.
Also, for a regular Spinner (non-hybrid), my non-merged branch have one which should work, if you want as a reference.